src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -84,7 +81,35 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(NeutralLanguage)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+  </Target>
+
+  <!-- 
+    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
+    create a hash of all assembly attributes so that the lock file name will change with the calculated
+    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate a new file.
+  -->
+  <Target Name="_CreateGeneratedAssemblyInfoInputsCacheFile"
+          DependsOnTargets="_CalculateAssemblyAttributes"
+          BeforeTargets="CoreGenerateAssemblyInfo">
```

Comment:
@dasMulli 

> Does the DependsOnTargets="A;B;C" set the build dependency of "b depends on a"?

Depends on how precise you're being with your language 😁

`DependsOnTargets="a;b;c"` does guarantee an order, unless there's another factor like `a DependsOnTargets="c"`. That happens because MSBuild uses a stack to order target execution, and encountering a target's `DependsOnTargets` pushes onto the stack in the given order. This is (almost) documented at https://docs.microsoft.com/en-us/visualstudio/msbuild/target-build-order.

There's another internal concept of dependency, though--when MSBuild considers a target and push prerequisites onto the stack, they get annotated with a "Parent". Then, if the target fails, the engine can additionally mark its parent as failed (and skip other targets that were building because of that parent). https://github.com/Microsoft/msbuild/pull/2133 is currently tweaking this for a corner case.

The latter is not done by an ordered `DependsOnTargets`--but that rarely matters.